core/pta/attestation.c

```diff
@@ -0,0 +1,600 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2020, Elias von Däniken
+ */
+
+#include <compiler.h>
+#include <stdio.h>
+#include <kernel/pseudo_ta.h>
+#include <mm/tee_pager.h>
+#include <mm/tee_mm.h>
+#include <pta_attestation.h>
+#include <string.h>
+#include <string_ext.h>
+#include <malloc.h>
+#include <tee_api_types.h>
+#include <crypto/crypto.h>
+#include <crypto/crypto_impl.h>
+
+#define RSA_SIG_SIZE_BYTE 256
+
+/*
+ * Attestation data is static to keep until the device boots again.
+ * Afterwards the attestation data has to be reloaded and everthing
+ * has to be attestated newly.
+ */
+static struct attestation_data {
+	uint8_t *iv;
+	uint32_t iv_size;
+	uint8_t *pub;
+	uint32_t pub_size;
+	uint8_t *enc_priv;
+	uint32_t enc_priv_size;
+	uint8_t *sig;
+	uint32_t sig_size;
+} att_data;
+
+static TEE_Result priv_asn1_decode(uint8_t *priv,
+				uint8_t **d, size_t *d_size)
+{
+
+	uint32_t size;
+	uint32_t offset = 7; // Beginn at the start of the modulus
+
+	// Check the header
+	if (priv[0] != 0x30)
+		return TEE_ERROR_SECURITY;
+
+	if (priv[1] != 0x82)
+		return TEE_ERROR_SECURITY;
+
+	// Get the size of the first integer aka modulus
+	if (priv[offset] != 0x02)
+		return TEE_ERROR_SECURITY;
+
+	if (priv[offset + 1] != 0x82)
+		return TEE_ERROR_SECURITY;
```

Comment:
```C
if (priv[0] != 0x30 || priv[1] != 0x82 || priv[offset] != 0x02 || priv[offset + 1] != 0x82)
	return TEE_ERROR_SECURITY;
```

What are the guarantees that accessing `priv[x]` is valid?

Reply:
Fix [here](https://github.com/OP-TEE/optee_os/pull/4011/commits/869e08962c7c36e9c532e5c6c3106410761fca39)